rcamp/tests/test_projects_commands.py

```diff
@@ -0,0 +1,259 @@
+from django.test import override_settings
+from django.conf import settings
+import mock
+import datetime
+import pytz
+
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import timezone
+
+from projects.models import (
+    Project,
+    Allocation,
+)
+from projects.management.commands.check_expiring_allocations import (
+    Command,
+    ExpirationNotifier,
+    UpcomingExpirationNotifier,
+)
+from tests.utilities.utils import (
+    SafeTestCase,
+)
+
+
+class ExpirationNotifierTestCase(SafeTestCase):
+
+    def setUp(self):
+        self.proj_dict = {
+            'project_id': 'ucb7',
+            'title': 'Test Project',
+            'description': 'A test project',
+            'pi_emails': ['testuser@test.org','cuuser@cu.edu'],
+            'organization':'ucb',
+        }
+        self.proj = Project.objects.create(**self.proj_dict)
+
+    def test_only_return_unnotified_expired_allocations(self):
+
+        sdate = datetime.datetime(2016,02,02)
+        sdate_tz = pytz.timezone('America/Denver').localize(sdate)
+        edate = datetime.datetime(2017,02,02)
+        edate_tz = pytz.timezone('America/Denver').localize(edate)
+
+        alloc_dict_not_notified = {
+            'project': self.proj,
+            'allocation_id': 'ucb7_summit2',
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_tz,
+            'expiration_notice_sent': False
+        }
+        alloc_not_notified = Allocation.objects.create(**alloc_dict_not_notified)
+
+        alloc_dict_notified = {
+            'project': self.proj,
+            'allocation_id': 'ucb6_summit2',
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_tz,
+            'expiration_notice_sent': True
+        }
+        alloc_notified = Allocation.objects.create(**alloc_dict_notified)
+
+        expiration_notifier = ExpirationNotifier()
+        actual_allocations = expiration_notifier.get_expired_allocations_with_no_sent_notification()
+        expected_query_set = [alloc_not_notified]
+
+        self.assertEqual(expected_query_set, list(actual_allocations))
+
+    def test_only_return_allocations_with_passed_expiration_date(self):
+
+        sdate = datetime.datetime(2016,02,02)
```

Comment:
Here and elsewhere: The date declarations should adjacent to where they are used, unless they are used multiple times within a block.

Reply:
It is worth making a helper function to localize a datetime, so that these declarations can be performed in one line throughout the test suite.